content/300-accelerate/250-connection-pooling.mdx

```diff
@@ -40,7 +40,7 @@ To change the size of the connection pool:
 
 ### Configuring the connection pool timeout
 
-The connection pool timeout is the duration, measured in seconds, during which the query engine must process a specific query; failing to do so within this time frame results in an exception being thrown, and the system proceeds to the next query in the queue.
+The connection pool timeout is the maximum number of seconds that Accelerate's internal connection pool waits for a new connection to your database to succeed and open. If the connection is not opened within this time frame, an exception is thrown. A value of 0 means there is no timeout.
```

Comment:
Slightly more nuance here 🙂 This isn't the time to establish the connection, but rather the time the query will wait in the queue for a free connection to be available.

```suggestion
The connection pool timeout is the maximum number of seconds that a query will block while waiting for a connection from Accelerate's internal connection pool. This occurs if the number of concurrent requests exceeds the connection limit, resulting in queueing of additional requests until a free connection becomes available. An exception is thrown if a free connection does not become available within the pool timeout. The connection pool timeout can be disabled by setting the value to 0.
```